src/trace_processor/perfetto_sql/stdlib/android/network_packets.sql

```diff
@@ -76,3 +79,163 @@ SELECT
 FROM __intrinsic_android_network_packets
 JOIN slice
   USING (id);
+
+-- This helper is used to unparenthesize a column list expression. Currently,
+-- the the pre-processor is unable to do both steps in one macro, so this macro
+-- must be passed to __intrinsic_token_apply at the callsite.
+CREATE PERFETTO MACRO __np_identity(
+    x Expr
+)
+RETURNS Expr AS
+$x;
+
+-- Finds groups of overlapping slices and assigns them group ids.
+--
+-- An overlap group is a set of slices (or instants) that contiguously have >=1
+-- slice present. For example, the following is one group:
+--
+--   ###   ##
+--     #####
+--
+-- Overlap detection is partitioned by the partition columns, where overlaps are
+-- only considered within the same partition.
+--
+-- Returns $src with two additional columns:
+-- * group_id: the group the row belongs to (per partition_columns)
+-- * max_end_so_far: the maximum end timestamp observed so far, useful for
+--   determining the time since the last group or event (per partition_columns)
+CREATE PERFETTO MACRO __add_overlap_group_id(
```

Comment:
Is this a variant of [1] (i.e. with n partition columns?).

Should we maybe put it there?

[1] https://source.corp.google.com/piper///depot/google3/third_party/perfetto/src/trace_processor/perfetto_sql/stdlib/intervals/overlap.sql;l=475?q=f:perfetto%20f:intervals%20f:overlap